mitiq/calibration/calibrator.py

```diff
@@ -92,208 +94,118 @@ def get_cost(self) -> Dict[str, int]:
+
+        return errors
+
+    def best_strategy(self, results: List[Result]) -> Strategy:
         """Finds the best strategy by using the parameters that had the
-        largest improvement factor.
+        smallest error.
 
         Args:
             results: Calibration experiment results. Obtained by first running
-                :func:`run_circuits` and :func:`compute_improvements`.
+                :func:`run_circuits`.
 
         Returns:
             A single :class:`Strategy` object specifying the technique and
             parameters that performed best.
         """
-        best_improvement_factor = 1.0
-        num_circuits = len(self.settings.circuit_types)
-
-        strategy = DefaultStrategy
-
-        def filter_on_strategy(
-            result: Dict[str, Dict[str, Dict[str, Any]]], strategy_id: int
-        ) -> Dict[str, Dict[str, Dict[str, Any]]]:
-            """Obtain results corresponding to the strategy of interest.
-            Args:
-                result: Calibration experiment results.
-                strategy_id: Index of the strategy of interest.
-
-            Returns:
-                A dictionary of results corresponding to the strategy of
-                interest.
-            """
-            res = result
-            res["mitigated_values"]["ZNE"]["results"] = [
-                res["mitigated_values"]["ZNE"]["results"][strategy_id]
-            ]
-            return res
-
-        for strategy_id in range(
-            len(results[0]["mitigated_values"]["ZNE"]["results"])
-        ):
-            strategy_group = []
-            for c in range(num_circuits):
-                result_copy = deepcopy(results)
-                strategy_group.append(
-                    filter_on_strategy(result_copy[c], strategy_id=strategy_id)
-                )
-                self.compute_improvements(strategy_group)
-
-            if (
-                strategy_group[0]["mitigated_values"]["ZNE"]["results"][0][
-                    "improvement_factor"
-                ]
-                > best_improvement_factor
-            ):
-                best_improvement_factor = strategy_group[0][
-                    "mitigated_values"
-                ]["ZNE"]["results"][0]["improvement_factor"]
-                strategy = strategy_group[0]["mitigated_values"]["ZNE"][
-                    "results"
-                ][0]["strategy"]
-        self.results.append(
-            {"best_improvement_factor": best_improvement_factor}
-        )
-
-        if strategy is DefaultStrategy:
-            warnings.warn("None of the improvement factors were > 1")
+        errors = self.compute_errors(results)
 
-        return strategy
+        best_strategy_id = min(errors, key=errors.__getitem__)
+        return self.get_strategy(best_strategy_id)
 
     def run(self) -> None:
         results = self.run_circuits()
-        self.compute_improvements(results)
         self.results = results
```

Comment:
`run()` seems now redundant with `run_circuits`

Reply:
Indeed it is now redundant. I've gone ahead and renamed `run_circuits -> run` where `run` stores the data internal to the class.